lib/rouge/lexers/varnish.rb

```diff
@@ -86,20 +96,47 @@ def self.variables
           push :inline_c
         end
 
-        rule %r/[a-z_.-]+/i do |m|
+        rule %r/\.?[a-z_][\w.-]*/i do |m|
           next token Keyword if self.class.keywords.include? m[0]
           next token Name::Function if self.class.functions.include? m[0]
           next token Name::Variable if self.class.variables.include? m[0]
           token Text
         end
 
-        # duration
-        rule %r/(?:#{LNUM}|#{DNUM})(?:ms|[smhdwy])/, Num::Other
-        # size in bytes
-        rule %r/#{LNUM}[KMGT]?B/, Num::Other
-        # literal numeric values (integer/float)
-        rule %r/#{LNUM}/, Num::Integer
-        rule %r/#{DNUM}/, Num::Float
+        ## for number literals
+
+        decimal = %r/[0-9]+/
+        hex = %r/[0-9a-f]+/i
+
+        numeric = %r{
+          (?:
+            0x#{hex}
+            (?:\.#{hex})?
+            (?:p[+-]?#{hex})?
+          )
+          |
+          (?:
+            #{decimal}
+            (?:\.#{decimal})?
+            (?:e[+-]?#{decimal})?
+          )
+        }xi
+
+        # duration literals
+        duration_suffix = Regexp.union(%w(ms s m h d w y))
+        rule %r/#{numeric}#{duration_suffix}/, Num::Other
+
+        # numeric literals (integer / float)
+        rule numeric do |m|
```

Comment:
I prefer such a "match and case-when" idiom here because VCL's number literals have so complex that defining everything in regexp is hard to read.